### PR TITLE
chore: Fix `cargo clippy` issues

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

**Comment by djc** [comment]
Are you sticking to nightly clippy for some reason? If so might make sense to have a CI job for that?

File: perf/src/bin/perf_client.rs [hunk 1]
```diff
@@ -308,7 +308,7 @@ async fn request(
 
     let send_stream_stats = stream_stats.new_sender(&send, upload);
 
-    const DATA: [u8; 1024 * 1024] = [42; 1024 * 1024];
+    static DATA: [u8; 1024 * 1024] = [42; 1024 * 1024];
```
Comment (djc) [inline perf/src/bin/perf_client.rs:311]: Why is this? I would usually assume const is better than static.

**Review by djc — APPROVED** [review]

File: perf/src/bin/perf_client.rs (hunk 1 above)
Comment (larseggert) [reply perf/src/bin/perf_client.rs:311]: https://rust-lang.github.io/rust-clippy/master/index.html#large_const_arrays

**Comment by larseggert** [comment]
> Are you sticking to nightly clippy for some reason? If so might make sense to have a CI job for that?

I just run the Nightly toolchain by default. CI would work but `clippy --fix` isn't super stable and so this usually requires manual interaction.

**Pull request merged**